CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -1444,10 +1448,85 @@ CF_EXPORT char **_CFEnviron(void) {
 }
 
 CF_CROSS_PLATFORM_EXPORT int _CFOpenFileWithMode(const char *path, int opts, mode_t mode) {
+#if TARGET_OS_WIN32
+    wchar_t *wide = createWideFileSystemRepresentation(path, NULL);
+
+    DWORD dwDesiredAccess = 0;
+    switch (opts & (O_RDONLY | O_WRONLY | O_RDWR)) {
+      case _O_RDONLY:
+        dwDesiredAccess = GENERIC_READ;
+        break;
+      case _O_WRONLY:
+        dwDesiredAccess = GENERIC_WRITE;
+        break;
+      case _O_RDWR:
+        dwDesiredAccess = GENERIC_READ | GENERIC_WRITE;
+        break;
+    }
+
+    DWORD dwCreationDisposition = OPEN_EXISTING;
```

Comment:
What happens if `_O_CREAT` was passed?  I think it is better to move L1475-L1477 here.